arangod/Cluster/SynchronizeShard.cpp

```diff
@@ -559,21 +559,21 @@ static arangodb::ResultT<SyncerId> replicationSynchronize(
     std::shared_ptr<DatabaseTailingSyncer> tailingSyncer,
     VPackBuilder& sy) {
 
-  auto& vocbase = col->vocbase();
-  auto database = vocbase.name();
+  std::shared_ptr<TRI_vocbase_t> vocbase(&col->vocbase());
```

Comment:
This is not safe. We cannot share ownership of a raw `TRI_vocbase_t*` with a `shared_ptr`.
For ex, if the last copy of the shared_ptr goes out of scope, it would delete the vocbase, although the original pointer is managed externally (by the DatabaseFeature and without shared_ptrs).